packages/monaco/src/browser/monaco-init.ts

```diff
@@ -50,6 +50,41 @@ import { MonacoQuickInputImplementation } from './monaco-quick-input-service';
 import { IQuickInputService } from '@theia/monaco-editor-core/esm/vs/platform/quickinput/common/quickInput';
 import { IStandaloneThemeService } from '@theia/monaco-editor-core/esm/vs/editor/standalone/common/standaloneTheme';
 import { MonacoStandaloneThemeService } from './monaco-standalone-theme-service';
+import { ContentHoverWidget } from '@theia/monaco-editor-core/esm/vs/editor/contrib/hover/browser/contentHoverWidget';
+import { IPosition } from '@theia/monaco-editor-core/esm/vs/editor/common/core/position';
+
+// VS Code uses 30 pixel for top height, and 24 pixels for bottom height, but Theia uses 32 pixel for the top and 22 for the bottom.
+// https://github.com/microsoft/vscode/blob/1430e1845cbf5ec29a2fc265f12c7fb5c3d685c3/src/vs/editor/contrib/hover/browser/resizableContentWidget.ts#L13-L14
+// https://github.com/eclipse-theia/theia/blob/b752ea690bdc4e7c5d9ab98a138504ead05be0d1/packages/core/src/browser/style/menus.css#L22
+// https://github.com/eclipse-theia/theia/blob/b752ea690bdc4e7c5d9ab98a138504ead05be0d1/packages/core/src/browser/style/status-bar.css#L18
+// https://github.com/eclipse-theia/theia/issues/14826
+function patchContentHoverWidget(actualTopHeight = 32): { setActualTopHeightForContentHoverWidget: (value: number) => void } {
+    const vscodeTopHeight = 30;
+    let _actualTopHeight = actualTopHeight;
+    function topHeightDiff(): number {
+        return _actualTopHeight - vscodeTopHeight;
+    }
+
+    const originalAvailableVerticalSpaceAbove = ContentHoverWidget.prototype['_availableVerticalSpaceAbove'];
+    ContentHoverWidget.prototype['_availableVerticalSpaceAbove'] = function (position: IPosition): number | undefined {
+        const value = originalAvailableVerticalSpaceAbove.call(this, position);
+        return typeof value === 'number' ? value - topHeightDiff() : undefined;
+    };
+
+    const originalAvailableVerticalSpaceBelow = ContentHoverWidget.prototype['_availableVerticalSpaceBelow'];
+    ContentHoverWidget.prototype['_availableVerticalSpaceBelow'] = function (position: IPosition): number | undefined {
+        const value = originalAvailableVerticalSpaceBelow.call(this, position);
+        return typeof value === 'number' ? value + 2 : undefined;
```

Comment:
Where is this 2 coming from? I assume it is the bottomHeightDiff as it steams from the 24 - 22? Shouldn't we also hand that in somehow from the outside like the top height? Also, I'm not too happy about the hard-coded 22, shouldn't we much rather determine this through `statusBar.node.getBoundingClientRect().height`?

Reply:
> Where is this 2 coming from? I assume it is the bottomHeightDiff as it steams from the 24 - 22?

Correct.

> Shouldn't we also hand that in somehow from the outside like the top height?

Theia can do this, but the bottom height doesn’t change. Why do you think it's a good idea to set this from the outside, for example, through the status bar (from the core package)? This would also introduce a back dependency.

> shouldn't we much rather determine this through statusBar.node.getBoundingClientRect().height?

Sure, Theia should calculate it from the status bar node, but this code cannot access the status bar instance without a query selector. I’m open to calculating it once and storing the client height, but I don't see dependency injection (DI) as a feasible option here.

Reply:
I think the bottom height can actually change quite easily. Theia exposes the `--theia-statusBar-height` CSS variable so adopters can very easily define another height for the status bar which should be considered correctly.

However, I do agree that dependency injection is definitely not a feasible option here. Much rather, it may make more sense to add an argument to `setActualTopHeightForContentHoverWidget` so that we can set the top AND the bottom height from the application shell where we do already have access to the status bar, the top panel, the tool bar, etc. In that case the method should probably drop the `Top` in the name. So basically I'm arguing that we should handle the bottom height the same way we handle the top height, what do you think?

Reply:
It's implemented as requested. Thanks for the suggestion.